torchbenchmark/models/torchrec_dlrm/__init__.py

```diff
@@ -0,0 +1,156 @@
+import torch
+
+# OSS import
+try:
+    # pyre-ignore[21]
+    # @manual=//ai_codesign/benchmarks/dlrm/torchrec_dlrm/data:dlrm_dataloader
+    from .data.dlrm_dataloader import get_dataloader
+except ImportError:
+    pass
+
+import itertools
+from pyre_extensions import none_throws
+from torchrec.datasets.criteo import DEFAULT_CAT_NAMES, DEFAULT_INT_NAMES
+from torchrec import EmbeddingBagCollection
+from torchrec.models.dlrm import DLRM, DLRM_DCN, DLRM_Projection, DLRMTrain
+from torchrec.modules.embedding_configs import EmbeddingBagConfig
+from torchrec.optim.apply_optimizer_in_backward import apply_optimizer_in_backward
+from torchrec.optim.keyed import CombinedOptimizer, KeyedOptimizerWrapper
+from torchrec.optim.optimizers import in_backward_optimizer_filter
+
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import RECOMMENDATION
+
+from .args import parse_args, InteractionType
+
+
+class Model(BenchmarkModel):
+    task = RECOMMENDATION.RECOMMENDATION
+    # upstream default batch size: 32
+    # https://github.com/facebookresearch/dlrm/blob/main/torchrec_dlrm/dlrm_main.py#L86
+    DEFAULT_TRAIN_BSIZE = 32
+    DEFAULT_EVAL_BSIZE = 32
+
+    def __init__(self, test, device, jit, batch_size=None, extra_args=[]):
+        super().__init__(test=test, device=device, jit=jit, batch_size=batch_size, extra_args=extra_args)
+        args = parse_args(self.extra_args)
+        backend = "nccl" if self.device == "cuda" else "gloo"
+        # initialize example data
+        if self.test == "train":
+            args.batch_size = self.batch_size
+            loader = get_dataloader(args, backend, "train")
+        if self.test == "eval":
+            args.test_batch_size = self.batch_size
+            loader = get_dataloader(args, backend, "test")
+        limit_batches = 1
+        iterator = itertools.islice(iter(loader), limit_batches)
+        self.example_inputs = next(iterator).to(self.device)
+        # parse the args
+        args.dense_arch_layer_sizes = [int(x) for x in args.dense_arch_layer_sizes.split(',') if x.strip().isdigit()]
```

Comment:
Need to parse the arg from `str` to `List[int]`, a bug in the original model file?